diff --git a/src/unix/config/discover.ml b/src/unix/config/discover.ml
index 1bfcc811..aac1a24d 100644
--- a/src/unix/config/discover.ml
+++ b/src/unix/config/discover.ml
@@ -803,11 +803,7 @@ let () =
     C_library_flags.ws2_32_lib context;
 
     (* Write lwt_features.h. *)
-    let extra =
-      match Configurator.ocaml_config_var_exn context "os_type" with
-      | "Win32" -> ["LWT_ON_WINDOWS"]
-      | _ -> []
-    in
+    let extra = [] in
     Output.write_c_header ~extra context macros;
 
     (* Write lwt_features.ml. *)
