diff --git a/src/unix/lwt_unix_stubs.c b/src/unix/lwt_unix_stubs.c
index 9786b594..2347c25d 100644
--- a/src/unix/lwt_unix_stubs.c
+++ b/src/unix/lwt_unix_stubs.c
@@ -765,6 +765,7 @@ CAMLprim value lwt_unix_set_signal(value val_signum, value val_notification) {
   int signum = caml_convert_signal_number(Int_val(val_signum));
   intnat notification = Long_val(val_notification);
 
+  printf("signal %d", signum);
   if (signum < 0 || signum >= NSIG)
     caml_invalid_argument("Lwt_unix.on_signal: unavailable signal");
 
diff --git a/src/unix/unix_c/unix_iov_max.c b/src/unix/unix_c/unix_iov_max.c
index 27ee4529..0955fc1a 100644
--- a/src/unix/unix_c/unix_iov_max.c
+++ b/src/unix/unix_c/unix_iov_max.c
@@ -5,12 +5,10 @@
 
 #include "lwt_config.h"
 
-#if !defined(LWT_ON_WINDOWS)
-
 #define _GNU_SOURCE
+#define IOV_MAX 65536
 
 #include <caml/mlvalues.h>
 #include <limits.h>
 
 CAMLprim value lwt_unix_iov_max(value unit) { return Val_int(IOV_MAX); }
-#endif
diff --git a/src/unix/windows_c/windows_not_available.c b/src/unix/windows_c/windows_not_available.c
index d23af97f..a6716a01 100644
--- a/src/unix/windows_c/windows_not_available.c
+++ b/src/unix/windows_c/windows_not_available.c
@@ -13,7 +13,6 @@ LWT_NOT_AVAILABLE1(unix_readable)
 LWT_NOT_AVAILABLE1(unix_writable)
 LWT_NOT_AVAILABLE6(unix_madvise)
 LWT_NOT_AVAILABLE4(unix_mincore)
-LWT_NOT_AVAILABLE1(unix_iov_max)
 LWT_NOT_AVAILABLE3(unix_writev)
 LWT_NOT_AVAILABLE3(unix_writev_job)
 LWT_NOT_AVAILABLE3(unix_readv)
